JSIL/AST/JSAstVisitor.cs

```diff
@@ -169,7 +169,7 @@ protected class Adapter<TVisitor, TNode>
         /// </summary>
         /// <param name="node">The node to visit.</param>
         /// <param name="name">The name to annotate the node with, if any.</param>
-        public void Visit (JSNode node, string name = null) {
+        public virtual void Visit (JSNode node, string name = null) {
```

Comment:
Making ```Visit``` virtual is nasty, this method gets called a *lot* by every single transform.

Since you're just using this to run a hook at the end of ```Visit``` for every node in ```AstEmitter```, could you move to a default-null delegate (or event) on ```AstVisitor``` that is called if not-null at the end of ```Visit```? That should be near-zero-cost for the other visitors.